qa/rpc-tests/mempool_size_limit_more.py

```diff
@@ -0,0 +1,287 @@
+        for s in range(NUM_NONCEASING_SIDECHAINS):
+            certs.append(load_from_storage(self.options.tmpdir + f"/certs/{s+NUM_CEASING_SIDECHAINS}/"))
+
+        return txs, ctxs, certs
+
+
+    def run_test(self):
+        mark_logs("Loading snapshot...", self.nodes, DEBUG_MODE, color = 'e')
+        txs, _, certs = self.setup_test() # We only need txs and certs
+
+        txs.sort(key=lambda x: x['feerate'])
+        for c in certs:
+            c.sort(key=lambda x: x['quality'])
+
+        mark_logs("Start from a clean mempool on node 0", self.nodes, DEBUG_MODE, color = 'e')
+        self.nodes[0].clearmempool()
+
+        mark_logs("Filling cert partition with certificates with fee = 0.00015 => feerate = F_low (~2.2465E-8)", self.nodes, DEBUG_MODE, color = 'c')
+        mark_logs("Trying to be conservative and not overcome cert partition limit ", self.nodes, DEBUG_MODE, color = 'c')
+        cert_fees = []
+        cert_sent_size, last_cert_size = 0, 0
+        maxfeerate, minfeerate = 0, MAX_FEE
+        while (cert_sent_size < NODE0_CERT_LIMIT_B - last_cert_size):
+            assert(len(certs[0]) > 0)
+            c = certs[0].pop(0)
+            cert_hex = self.nodes[0].sendrawtransaction(c['hex'])
+            last_cert_size = len(c['hex']) // 2
+            cert_fees.append(c['feerate'])
+            cert_sent_size += last_cert_size
+            minfeerate = min(minfeerate, c['feerate'])
+            maxfeerate = max(maxfeerate, c['feerate'])
+
+        mark_logs(cc('g', "certificates -- minfeerate: ") + f"{minfeerate:.8E}" + cc('g', " - maxfeerate: ") + f"{maxfeerate:.8E}", self.nodes, DEBUG_MODE)
+        mempoolstatus = self.nodes[0].getmempoolinfo()
+        mark_logs(cc('g', "Current mempool state: bytes: ") + f"{mempoolstatus['bytes']}" +
+                  cc('g', " - bytes-for-tx: ") + f"{mempoolstatus['bytes-for-tx']}" +
+                  cc('g', " - bytes-for-cert: ") + f"{mempoolstatus['bytes-for-cert']}", self.nodes, DEBUG_MODE)
+        certPartitionStatus = int(mempoolstatus['bytes-for-cert'])
+        print()
+
+
+        mark_logs("Filling tx partition with transactions having feerate > 0.00015 (F_high)", self.nodes, DEBUG_MODE, color = 'c')
+        mark_logs("Trying to be conservative and not overcome tx partition limit ", self.nodes, DEBUG_MODE, color = 'c')
+        self.utxos = self.nodes[0].listunspent()
+        tx_sent = 0
+        last_tx_size = 0
+        mintxfeerate, maxtxfeerate = MAX_FEE, 0
+        discarded_txs = 0
+        usage = self.nodes[0].getmempoolinfo()
+        while(int(usage['bytes-for-tx']) < NODE0_LIMIT_B // 2 - last_tx_size and int(usage['bytes']) < NODE0_LIMIT_B - last_tx_size):
+            if len(txs) > 0:
+                tx = txs.pop(0)
+            else:
+                assert(len(self.utxos) > 0)
+                tx = self.create_a_big_transaction(self.utxos)
+            if tx['feerate'] <= Decimal('0.00015'):
```

Comment:
Given that we know the size of `tx`, we might make the test more robust by executing `continue` also if its size would result in a rejection.

Reply:
(as previous response)